Controllers/Services/Email/EmailService.cs

```diff
@@ -16,24 +20,61 @@ public EmailService(IOptions<EmailServiceOptions> options)
         {
             FromName = options.Value.FromName;
             FromAddress = options.Value.FromAddress;
+            ToName = options.Value.ToName;
+            ToAddress = options.Value.ToAddress;
             SmtpServer = options.Value.SmtpServer;
             SmtpPort = options.Value.SmtpPort;
         }
 
-        public void SendEmail(
-            string toName, string toAddress,
-            string subject, string text
-        )
+        public void SendTaskResultEmail(EmployeeTaskResult taskResult)
         {
+            try
+            {
+                SendEmail(
+                    MessageHelper.EmailSubjectFromTaskAndOutcome(
+                        taskResult.Task,
+                        taskResult.TaskOutcome
+                    ),
+                    taskResult.Message
+                );
+            }
+            catch
+            {
+                // Fail silently; this is currently offered on a best-effort
+                // basis.
+            }
+        }
+
+        public void SendFailureEmail(TaskEnum task, Exception e)
+        {
+            try
+            {
+                SendEmail(
+                    MessageHelper.EmailSubjectFromTaskAndOutcome(task, TaskOutcomeEnum.Fail),
+                    MessageHelper.MessageFromException(e)
+                );
+            }
+            catch
+            {
+                // Fail silently; this is currently offered on a best-effort
+                // basis.
+            }
+        }
+
+        protected void SendEmail(string subject, string text)
+        {
+            // This is a no-op if either ToName and ToAddress are null or blank
+            if (ToName == null || ToName.Equals("") || ToAddress == null || ToAddress.Equals(""))
+            {
+                return;
```

Comment:
Would be great to log a message here, saying an email address / name isn't provided.